strapi/strapi
# CM: Fix i18n LocalePicker component

### What does it do?

Fixes a rendering-order issue in the CM (and a corresponding error overlay), if users switch between locales. We know [about the issue](https://github.com/strapi/strapi/issues/14225) for quite some time and I suspected this error to be in the design-system, but I don't see how [`source`](https://github.com/strapi/design-system/blob/7da56a87f203fd228f2e482f47aa6da0b25b75a3/packages/strapi-design-system/src/Popover/Popover.js#L10) could ever be undefined in the `Select` component itself and therefore I think this is a problem with how the component re-renders.

| Before | After |
|-|-|
| <video src="https://user-images.githubusercontent.com/2244375/187145230-ccfe30ae-b18d-4ab9-991c-f62d0abe6ae2.mp4" /> | <video src="https://user-images.githubusercontent.com/2244375/187145241-cd84cef4-457f-45a1-85f2-8ddfbe2a5c9a.mp4" /> |

### Why is it needed?

Fixes an error the is visible during development.

### How to test it?

1. Navigate to the CM list view in *development mode*
2. Use the kitchen-sink example
3. switch locales: you should not see the error overlay

### Related issue(s)/PR(s)

- https://github.com/strapi/strapi/issues/14225
